### PR TITLE
New ingredient analysis for vegan / vegetarian / palm oil,

### PR DESCRIPTION
**Pull request by stephanegigandet**
issue #1930 #522 #1052

Start of the new ingredient classes parsing for vegan / vegetarian / palm oil
Start of the display
New /ingredients-analysis facet

### CODE REVIEW

File: po/common/common.pot [hunk 1]
```diff
@@ -3436,3 +3436,14 @@ msgctxt "popularity_p"
...
 
+msgctxt "ingredients_analysis_p"
+msgid "ingredients analysis"
+msgstr ""
+
+msgctxt "ingredients_analysis_s"
+msgid "ingredients analysis"
+msgstr ""
+
+msgctxt "ingredients_analysys"
```
Comment (hangy) [inline po/common/common.pot]: I don't think "analysys" is an existing word. This translation seems to be unused?

File: po/common/en.po [hunk 2]
```diff
@@ -3274,3 +3274,16 @@ msgstr "Collapsed changes"
...
+
+msgctxt "ingredients_analysis_p"
+msgid "ingredients analysis"
+msgstr "ingredients analysis"
+
+msgctxt "ingredients_analysis_s"
+msgid "ingredients analysis"
+msgstr "ingredients analysis"
+
+msgctxt "ingredients_analysys"
```
Comment (hangy) [inline po/common/en.po]: I don't think "analysys" is an existing word. This translation seems to be unused?

File: t/tags.t [hunk 3]
```diff
@@ -5,7 +5,7 @@ use Modern::Perl '2012';
 use utf8;
 
 use Test::More;
-use Log::Any::Adapter 'TAP', filter => "none";;
+#use Log::Any::Adapter 'TAP', filter => "none";;
```
Comment (hangy) [inline t/tags.t]: Why did you remove this?

File: taxonomies/ingredients_analysis.txt [hunk 4]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (hangy) [inline taxonomies/ingredients_analysis.txt]: Could you also add this new taxonomy to `refresh_taxonomies.js`, so that Travis CI can rebuild it?

Comment (stephanegigandet) [reply taxonomies/ingredients_analysis.txt]: Sure.

File: t/tags.t (hunk 3 above)
Comment (stephanegigandet) [reply t/tags.t]: Just to remove some logs in the output while I was testing. I can put it back.

File: po/common/en.po (hunk 2 above)
Comment (stephanegigandet) [reply po/common/en.po]: Good catch. On the dev server vi has weird colors that I haven't figured out how to remove yet. It makes typos too easy.

**Comment by stephanegigandet** [comment]
Silly me, I had forgotten to check in the actual code for the ingredients analysis in Ingredients.pm :-)

File: lib/ProductOpener/Display.pm [hunk 5]
```diff
@@ -1463,6 +1463,13 @@ sub display_list_of_tags($$) {
 			unknown_tags_products => 0,
 		);
 
+		my $missing_property = $request_ref->{missing_property};
+		if ((defined $missing_property) and ($missing_property !~ /:/)) {
+			$missing_property .= ":en";
+			$log->debug("missing_property defined", {missing_property => $missing_property});
+			sleep(5);
```
Comment (hangy) [inline lib/ProductOpener/Display.pm]: `sleep` in production?

Comment (stephanegigandet) [reply lib/ProductOpener/Display.pm]: :-) I'll remove it.
